integration_cleanup/aws_cleanup.py

```diff
@@ -0,0 +1,190 @@
+import boto3
+
+
+class AwsCleanupInstance:
+    """
+    Handles the cleanup of IAM users, Policies and Buckets related to CircleCI
+    """
+    def __init__(self, age_limit, aws_access_key_id, aws_secret_access_key,
+                 policy_name, dry_run=False):
+        """
+        Set's up AWS connections and clients
+        """
+        self.age_limit = age_limit
+        self.dry_run = dry_run
+        self.policy_name = policy_name
+        self.cleaned_up_hashes = []
+
+        self.session = boto3.Session(
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+        )
+        self.iam_client = self.session.client('iam')
+        self.s3_client = self.session.client('s3')
+        self.s3_resource = self.session.resource('s3')
+
+    def delete_bucket(self, bucket_name):
+        """
+        Deletes a S3 bucket and all of it's files
+        """
+        try:
+            print("Deleting bucket {}.".format(bucket_name))
+            bucket = self.s3_resource.Bucket(bucket_name)
+            bucket.object_versions.delete()
+            bucket.objects.all().delete()
+            bucket.delete()
+        except self.s3_client.exceptions.NoSuchBucket:
```

Comment:
@giovannicimolin, is this the only exception to be handled? In Ocim code, we handle the `boto.exception.S3ResponseError` while trying to delete a bucket. Can you check and add a handler + logging statement to log the error?

Reply:
I've put `botocore.exceptions.ClientError` to be a little more generic when catching errors relaetd to the deletion of a bucket